packages/flutter/lib/src/material/tooltip.dart

```diff
@@ -114,11 +144,100 @@ class Tooltip extends StatefulWidget {
     this.child,
     this.triggerMode,
     this.enableFeedback,
-  }) : assert(message != null),
+  }) : assert(
+         message != null,
+         'A non-null String must be provided in message to a Tooltip widget.',
+       ),
+       tooltip = null,
+       semanticsLabel = null,
        super(key: key);
 
-  /// The text to display in the tooltip.
-  final String message;
+  /// Creates a tooltip using a widget provided to the [tooltip].
+  ///
+  /// **NOTE :** Using [Tooltip] and [Tooltip.custom], like examples given
+  /// below, will show the same results unless the custom [textStyle] or
+  /// [TooltipTheme] is not procvided for the [Tooltip]. Therefore, it is
+  /// recommended to use the [Tooltip] instead of [Tooltip.custom] in general
+  /// scenarios where just simple text tooltips needed like following examples.
+  /// ```dart
+  /// Tooltip(
+  ///   message:"my tooltip",
+  /// )
+  ///
+  /// Tooltip.custom(
+  ///   tooltip: Text("my tooltip"),
+  /// )
+  /// ```
+  ///
+  /// **NOTE :**
+  /// By default, tooltips should adhere to the
+  /// [Material specification](https://material.io/design/components/tooltips.html#spec).
+  /// But, in the case of [Tooltip.custom], this will ignore the [TooltipTheme.of] as widget
+  /// provided in the [tooltip] will use themes like as it would be using in the
+  /// normal application.
+  ///
+  /// All parameters that are defined in the constructor will
+  /// override the default values _and_ the values in [TooltipTheme.of].
+  const Tooltip.custom({
+    Key? key,
+    required Widget this.tooltip,
+    this.height,
+    this.padding,
+    this.margin,
+    this.verticalOffset,
+    this.preferBelow,
+    this.semanticsLabel,
+    this.decoration,
+    this.waitDuration,
+    this.showDuration,
+    this.child,
+    this.triggerMode,
+    this.enableFeedback,
+  }) : assert(
+         tooltip != null,
+         'A non-null Widget must be provided in tooltip to a Tooltip.custom widget.',
+       ),
+       message = null,
+       textStyle = null,
+       excludeFromSemantics = true,
```

Comment:
Yes, I did it intentionally as the message is used for tooltip as well as for accessibility, and when we make excludeFromSemantics true it will not add the semantics for the tooltip. Same way, I made it semanticsLabel nullable. Therefore, if it is null, it won't add any semantics. But when provided, it will be added and used for accessibility.

Reply:
Yes, I understand how it works, but I still think it is confusing. If someone sees this line in the code, they might not check the code further on that ignores this setting. Also, introducing two different ways to do the same thing usually confuses things.